tests/ui/UnreadIndicatorsTest.tsx

```diff
@@ -520,30 +518,38 @@ describe('Unread Indicators', () => {
                 return waitFor(() => expect(isNewMessagesBadgeVisible()).toBe(false));
             }));
 
-    it('Keep showing the new line indicator when a new message is created by the current user', () =>
-        signInAndGetAppWithUnreadChat()
-            .then(() => {
-                // Verify we are on the LHN and that the chat shows as unread in the LHN
-                expect(areYouOnChatListScreen()).toBe(true);
-
-                // Navigate to the report and verify the indicator is present
-                return navigateToSidebarOption(0);
-            })
-            .then(async () => {
-                await act(() => transitionEndCB?.());
-                const newMessageLineIndicatorHintText = Localize.translateLocal('accessibilityHints.newMessageLineIndicator');
-                const unreadIndicator = screen.queryAllByLabelText(newMessageLineIndicatorHintText);
-                expect(unreadIndicator).toHaveLength(1);
+    it('Keep showing the new line indicator when a new message is created by the current user', async () => {
+        await signInAndGetAppWithUnreadChat();
+
+        // Verify we are on the LHN and that the chat shows as unread in the LHN
+        expect(areYouOnChatListScreen()).toBe(true);
+
+        // Navigate to the report and verify the indicator is present
+        await navigateToSidebarOption(0);
+        await act(() => transitionEndCB?.());
+        let newMessageLineIndicatorHintText = Localize.translateLocal('accessibilityHints.newMessageLineIndicator');
+        let unreadIndicator = screen.queryAllByLabelText(newMessageLineIndicatorHintText);
+        expect(unreadIndicator).toHaveLength(1);
+
+        // Leave a comment as the current user and verify the indicator is not removed
+        const reportActionsBefore = (await TestHelper.onyxGet(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${REPORT_ID}`)) as Record<string, ReportAction>;
```

Comment:
Lines 535 -> 547 are the most important changes in this test file - the rest of it is just using `async`/`await` instead of promise chains without further changes.

These lines simulate a server response setting a correct value for `previousReportActionID` when our action is finalized.

It's not 100% clear why this wasn't needed before, but it seems like using `useOnyx` results in faster re-renders in a test environment, and that test was a false positive before just testing the content we had on the screen before calling `Report.addComment`.

Reply:
Updated the test to verify the new comment is visible on the screen after it's added, to prevent any possible false positives in the future.